controllers/workspace/http.go

```diff
@@ -63,4 +72,46 @@ func setupHttpClients() {
 		Transport: healthCheckTransport,
 		Timeout:   500 * time.Millisecond,
 	}
+	InjectCertificates(k8s, logger)
+}
+
+func InjectCertificates(k8s client.Client, logger logr.Logger) {
+	if certs, ok := readCertificates(k8s, logger); ok {
+		for _, certsPem := range certs {
+			injectCertificates([]byte(certsPem), httpClient.Transport.(*http.Transport), logger)
+		}
+	}
+}
+
+func readCertificates(k8s client.Client, logger logr.Logger) (map[string]string, bool) {
+	configmapRef := config.GetGlobalConfig().Routing.TLSCertificateConfigmapRef
+	if configmapRef == nil {
+		return nil, false
+	}
+	configMap := &corev1.ConfigMap{}
+	namespacedName := &types.NamespacedName{
+		Name:      configmapRef.Name,
+		Namespace: configmapRef.Namespace,
+	}
+	err := k8s.Get(context.Background(), *namespacedName, configMap)
+	if err != nil {
+		logger.Error(err, "Failed to read configmap with certificates")
+		return nil, false
+	}
+	return configMap.Data, true
+}
+
+func injectCertificates(certsPem []byte, transport *http.Transport, logger logr.Logger) {
+	caCertPool := transport.TLSClientConfig.RootCAs
+	if caCertPool == nil {
+		systemCertPool, err := x509.SystemCertPool()
+		if err != nil {
+			logger.Error(err, "Failed to load system cert pool")
+			caCertPool = x509.NewCertPool()
+		}
+		caCertPool = systemCertPool
```

Comment:
This line needs to be in an `else` block so that `caCertPool` will only be set to `systemCertPool` if there was not error in retrieving the SystemCertPool (`x509.SystemCertPool()` will return a nil value and an error if an error occurs).

```go
	systemCertPool, err := x509.SystemCertPool()
	if err != nil {
		logger.Error(err, "Failed to load system cert pool")
		caCertPool = x509.NewCertPool()
	} else {
	        caCertPool = systemCertPool
	} 

```

Reply:
Yes, thank you for the catch, fixed.